lambdas/reconcile-migrated-vcs/src/main/java/uk/gov/di/ipv/core/reconcilemigratedvcs/ReconcileMigratedVcsHandler.java

```diff
@@ -136,10 +138,14 @@ public ReconciliationReport handleRequest(Request request, Context context) {
             if (e instanceof InterruptedException) {
                 Thread.currentThread().interrupt();
             }
-            LOGGER.error(
+            var logMessage =
                     LogHelper.buildErrorMessage("Parallel execution failed", e)
                             .with(LOG_BATCH_ID.getFieldName(), reconciliationReport.getBatchId())
-                            .with(LOG_ERROR_STACK.getFieldName(), e.getStackTrace()));
+                            .with(LOG_ERROR_STACK.getFieldName(), e.getStackTrace());
+            if (e.getCause() != null && e.getCause().getStackTrace() != null) {
+                logMessage.with("causeErrorStack", e.getCause().getStackTrace());
```

Comment:
This is just adding logging a stack trace for the cause - otherwise the stacktrace for an ExecutionException just takes you to the `get()` call on the task (line 130)

Reply:
This feels like a bit of a smell tbh, but I'm willing to leave it in just for this script.

Ideally we'd leverage the error logging capability within Log4J (which must be able to log stack traces sensibly), and if that's not possible then use something like `printStackTrace`.

For example this won't handle more than one nested `cause`...